tensorvis/tensorvisbase/include/inviwo/tensorvisbase/datastructures/tensorfield.h

```diff
@@ -0,0 +1,441 @@
+#pragma once
+
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/datastructures/buffer/buffer.h>
+#include <inviwo/core/datastructures/buffer/bufferramprecision.h>
+#include <inviwo/core/datastructures/datamapper.h>
+#include <inviwo/core/datastructures/image/image.h>
+#include <inviwo/core/util/indexmapper.h>
+#include <inviwo/tensorvisbase/tensorvisbasemoduledefine.h>
+#include <inviwo/tensorvisbase/util/tensorutil.h>
+#include <inviwo/tensorvisbase/util/misc.h>
+#include <Eigen/Dense>
+#include <inviwo/core/datastructures/spatialdata.h>
+#include <inviwo/tensorvisbase/datastructures/attributes.h>
+#include <inviwo/dataframe/datastructures/dataframe.h>
+#include <inviwo/dataframe/datastructures/column.h>
+#include <type_traits>
+#include <optional>
+#include <memory>
+#include <inviwo/core/util/constexprhash.h>
+
+namespace inviwo {
+/**
+ * \class TensorField
+ * \brief Base data structure for tensorfields.
+ */
+template <unsigned int N, typename precision>
+class TensorField : public StructuredGridEntity<N> {
+public:
+    using value_type = precision;
+    using sizeN_t = glm::vec<N, std::size_t, glm::defaultp>;
+    using matN = glm::mat<N, N, precision, glm::defaultp>;
+    using matNb = glm::mat<N + 1, N + 1, precision, glm::defaultp>;
+    using vecN = glm::vec<N, precision, glm::defaultp>;
+
+    TensorField() = delete;
+
+    TensorField(const sizeN_t& dimensions, const std::vector<matN>& tensors);
+    TensorField(const sizeN_t& dimensions, std::shared_ptr<std::vector<matN>> tensors);
+
+    TensorField(const sizeN_t& dimensions, const std::vector<matN>& tensors,
```

Comment:
would it be possible to merge these ones with the above ones?

Reply:
Done.